dbf2csv/models/dbf2csv.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 INGETIVE (<https://ingetive.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import csv
+import base64
+from odoo import models, fields, api
+from dbfpy import dbf
+
+
+class FileDBF(models.TransientModel):
+    _name = 'dbftocsv.filedbf'
+    _description = 'File DBF to convert to CSV'
+
+    data_file = fields.Binary(
+        string='DBF file',
+        required=True,
+        help='Get you DBF file.')
+    filename = fields.Char(string='DBF file name')
+
+    @api.multi
+    def proccess_file(self):
+        # Process the DBF file chosen in the wizard,
+        # create and download CSV file.
+        self.ensure_one()
+        # Create DBF file in TMP
+        fp = open('/tmp/' + self.filename, 'w+')
```

Comment:
I would suggest to avoid writting to external files (eg. by adding additional binary field, like I've mentioned before) as `self.filename` could be manipulate to overwrite system files writable by the Odoo user. 

At least this can be improved by using [`tempfile`](https://docs.python.org/2/library/tempfile.html#tempfile.NamedTemporaryFile), eg.:

```python
with tempfile.NamedTemporaryFile(mode='w+') as tmp_csv_file:
    # ...
```
It provides the benefit that it deletes the temporary file after it is closed (avoids spamming the file system) and is more failsafe - as it creates the file in a temporary directory (dependent on OS).

Reply:
Also, IIUC `dbf.Dbf` [supports reading from a file-like object](http://dbfpy.bzr.sourceforge.net/bzr/dbfpy/annotate/head:/dbfpy/dbf.py#L96), so I would suggest to rewrite the module using in-memory file objects ([`io`](https://docs.python.org/2/library/io.html), [`(c)StringIO`](https://docs.python.org/2/library/stringio.html)).